### PR TITLE
Make components reusable

### PR DESCRIPTION
**Pull request by ninetteadhikari**
## What is the Purpose?
- Cascader: remove unused `methods` and `data`
- Checkbox: remove unused `data` and props, expose data array
- Input: add `placeholder` as props
- Navbar: remove unused `methods`
- Switch: add v-model value as props

## What was the approach?
Add props and remove unused methods

## Are there any concerns to addressed further before or after merging this PR?
None

## Mentions?
@andrewtpham @hvitis 

## Issue(s) affected?
None

### CODE REVIEW

File: src/components/checkbox/CheckboxGroup.vue [hunk 1]
```diff
@@ -11,16 +11,13 @@ export default {
     type: {
```
Comment (andrewtpham) [inline src/components/checkbox/CheckboxGroup.vue:11]: If `:type="type"` is removed, can we remove these lines 11-13?

Comment (lupyana) [reply src/components/checkbox/CheckboxGroup.vue:11]: @andrewtpham Yes, thay can be removed, they are not used in that file.

Comment (lupyana) [reply src/components/checkbox/CheckboxGroup.vue:11]: Also according to the [element](https://element.eleme.io/#/en-US/component/checkbox) docs, checkbox group component does not have a type as a prop.

Comment (ninetteadhikari) [reply src/components/checkbox/CheckboxGroup.vue]: Thanks both :) I have removed the `type` props

**Pull request merged**